From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Thu, 27 Nov 2025 03:57:18 -0500
Subject: [PATCH 27/46] 
 0036-win32u-Pass-appropriate-thread-id-for-KEYEVENTF_SCAN.patch

---
 dlls/win32u/message.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/dlls/win32u/message.c b/dlls/win32u/message.c
index b6cdd3efa0e..7b644818250 100644
--- a/dlls/win32u/message.c
+++ b/dlls/win32u/message.c
@@ -3937,9 +3937,7 @@ NTSTATUS send_hardware_message( HWND hwnd, UINT flags, const INPUT *input, LPARA
             if (input->ki.dwFlags & KEYEVENTF_SCANCODE)
             {
                 UINT scan = input->ki.wScan;
-                /* TODO: Use the keyboard layout of the target hwnd, once
-                 * NtUserGetKeyboardLayout supports non-current threads. */
-                HKL layout = NtUserGetKeyboardLayout( 0 );
+                HKL layout = NtUserGetKeyboardLayout(NtUserGetWindowThread( hwnd, NULL ));
                 if (flags & SEND_HWMSG_INJECTED)
                 {
                     scan = scan & 0xff;
